### ISSUE TITLE
Aim offset appears to use the wrong data in part of its calculation

### ISSUE THREAD

**Issue by mrbelowski** [praydog/UEVR#169]
https://github.com/praydog/UEVR/blob/6e8457ccb82f8006e262503128d5233552cc1749/src/mods/vr/IXRTrackingSystemHook.cpp#L1770

Where we're controller-aiming we adjust the aim point slight to take into account the offset between the HMD (the camera) and the controller (the weapon). I think this is using the wrong source of data for the offset. It manifests as an offset between the aim point and the impact point (or for games with an aim reticle, the reticle and the gun iron sight). This offset varies as you rotate in your play space.

I think this happens because the offset calc uses the standing origin, which is fixed, then applies this to a frame of reference that's rotating (the current camera orientation). Or something like that.

TL/DR - I think the above line should be

const auto adjusted_forward = glm::normalize(right_controller_end - glm::vec3{vr->get_position(0)});

where get_position(0) is returning the current HMD position, rather than the previously set standing origin

**Comment by praydog** [comment]
First glance, I don't think so - because the "standing origin" is where the player is originally standing. This automatically gets recentered with "roomscale movement" on. And when it's not on, the "standing origin" is where the player actually is in the game world.

**Comment by praydog** [comment]
If you can provide some evidence of any issues here - I am happy to take a second look at this code.

**Comment by mrbelowski** [comment]
Perhaps I'm misunderstanding the purpose of this adjustment. I'm imagining a triangle with the short side being the horizontal distance between hmd and controller, and the long side being 10 metres out from this controller end. Them the adjacent angle is used to offset the aim.

If that's the case, don't we need the current hmd orientation to derive the horizontal distance, not the standing origin?  The origin is only valid here if we set it on each tick, otherwise the horizontal distance component changes as you rotate - if you happen to be facing the same direction as your origin direction then it works. If you're rotated 90 degrees then all that horizontal offset ends up along the z axis instead so the adjustment angle becomes zero.

But you know your code and maths and I'm still trying to get my head around it. I'll try and repro the original issue that lead me here and if I can I'll post a video. Thanks for.gerting back to me :)

**Comment by mrbelowski** [comment]
i tested this back to back last night and you're quite right, the change I'm suggesting doesn't have a positive impact (or any impact at all as far as I can see). I'll close this and go back to just having fun
